### PR TITLE
Replace global styling registry with reducer

### PR DESCRIPTION
**Pull request by edgarmueller**
Also:
- Change signature of initJsonFormsStore to expect single state object
- Replace dot-prop-immutable with lodash API
- Remove warnings

### CODE REVIEW

File: packages/core/src/renderers/renderer.util.tsx [hunk 1]
```diff
@@ -146,19 +143,20 @@ export const mapStateToControlProps = (state, ownProps) => {
...
   if (trim) {
-    classNames = classNames.concat(JsonForms.stylingRegistry.get('control.trim'));
+    classNames = classNames.concat(getStyle(state)('control.trim'));
   }
   const inputClassName =
     ['validate']
       .concat(isValid ? 'valid' : 'invalid');
-  const labelClass = JsonForms.stylingRegistry.getAsClassName('control.label');
+  const labelClass = getStyleAsClassName(state)('control.label');
+  const descriptionClassName = getStyleAsClassName(state)('input-description');
```
Comment (eneufeld) [inline packages/core/src/renderers/renderer.util.tsx]: I think this should be `control.description`

File: packages/core/src/renderers/renderer.util.tsx [hunk 2]
```diff
@@ -54,31 +56,26 @@ export const mapStateToLayoutProps = (state, ownProps) => {
   return {
     renderers: state.renderers,
     visible,
-    path: ownProps.path
+    path: ownProps.path,
+    getStyle: getStyle(state),
```
Comment (eneufeld) [inline packages/core/src/renderers/renderer.util.tsx]: this is vanilla renderer specific, maybe this should be handled similar to the vanillacontrolprops ?

File: packages/core/src/actions.ts [hunk 3]
```diff
@@ -59,3 +63,19 @@ export const unregisterRenderer = (
   tester,
   renderer
 });
+
+export const registerStyle = (styleName: string, classNames: string[]) => ({
```
Comment (eneufeld) [inline packages/core/src/actions.ts]: can we make this specific to vanilla?

File: packages/core/src/renderers/renderer.util.tsx [hunk 4]
```diff
@@ -54,31 +56,26 @@ export const mapStateToLayoutProps = (state, ownProps) => {
...
+    getStyleAsClassName: getStyleAsClassName(state)
   };
 };
 
 export const renderChildren = (
```
Comment (eneufeld) [inline packages/core/src/renderers/renderer.util.tsx]: I think we should move the whole method to the vanilla renderer set

File: packages/core/src/renderers/renderer.util.tsx [hunk 5]
```diff
@@ -54,31 +55,24 @@ export const mapStateToLayoutProps = (state, ownProps) => {
   return {
     renderers: state.renderers,
     visible,
-    path: ownProps.path
+    path: ownProps.path,
   };
 };
 
 export const renderChildren = (
```
Comment (eneufeld) [inline packages/core/src/renderers/renderer.util.tsx]: this is vanilla dependent now as material is using its own implementation

File: packages/material/src/controls/material-input.control.tsx [hunk 6]
```diff
@@ -31,15 +33,18 @@ export class MaterialInputControl extends Control<ControlProps, ControlState> {
     if (!visible) {
       style = {display: 'none'};
     }
+    const classNames: string[] = !_.isEmpty(controlElement.scope) ?
```
[the remark this answers is not in the document]
Comment (edgarmueller) [reply packages/material/src/controls/material-input.control.tsx]: Material input tests are expecting this class in order to retrieve controls to assert upon. Of course, we could also update the test cases, I guess.

**Pull request merged**